bitnami/thanos/values.yaml

```diff
@@ -1781,6 +1781,30 @@ ruler:
   ##
   alertmanagers: []
 
+  ## Alertmanagers Configuration (passed directly to the `ruler`)
+  ## If `file` is given intead of an alertmanager config, the file
+  ## will be passed directly to the thanos ruler, but no secret or
+  ## volume mount will be created
+  ##
+  ## Example 1:
+  ## # Assumes the alertmanagers_config.yaml is mounted to the pod
+  # alertmanagersConfig:
+  #  file: alertmanagers_config.yaml
```

Comment:
`alertmanagersConfig.file` is for those who want to bring their own configuration file, or want to reuse one that is already in the cluster

`alertmanagersConfig.alertmanagers` covers those who want a secret created for them and mounted to the pod.

Reply:
Could you explain a litter more about the secret?

Reply:
👍  the alertmanager config can contain potentially secret values like basic auth passwords or bearer tokens, so we take the passed configuration, store it into a secret and then mount that to a pod (you can read more about the configuration and potential secrets here: https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md#alertmanager)